src/member/member.service.ts

```diff
@@ -1,92 +1,91 @@
       .map((member) => ({
         userId: member.userId,
-        name: member.user.name,
+        name: member.user?.name ?? '',
         role: member.role,
         joinedAt: member.createdAt,
       }));
   }
 
-  async getGroupMember(groupId: number, userId: number) {
-    return this.prisma.userGroup.findFirst({
-      where: {
-        groupId,
-        userId,
-      },
-      include: {
-        user: true,
-      },
-    });
+  async getGroupMember(
+    groupId: number,
+    userId: number,
+  ): Promise<MemberResponseDto | null> {
+    const member = await this.memberRepository.findGroupMember(groupId, userId);
+    if (!member) {
+      return null;
+    }
+    return {
+      userId: member.userId,
+      name: member.user?.name ?? '',
+      role: member.role,
+      joinedAt: member.createdAt,
+    };
   }
 
-  async validateRemoveMember(
+  async processRemoveMember(
     groupId: number,
     targetUserId: number,
-    requesterUserId: number,
-  ) {
-    if (targetUserId === requesterUserId) {
-      throw new ForbiddenException('자신을 삭제할 수 없습니다.');
-    }
-
-    const requester = await this.prisma.userGroup.findFirst({
-      where: {
-        groupId,
-        userId: requesterUserId,
-      },
-    });
-
-    if (!requester || requester.role !== 'manager') {
-      throw new ForbiddenException('매니저만 멤버를 삭제할 수 있습니다.');
-    }
-
-    const targetMember = await this.prisma.userGroup.findFirst({
-      where: {
-        groupId,
-        userId: targetUserId,
-      },
-    });
-
+  ): Promise<MemberResponseDto> {
+    const targetMember = await this.memberRepository.findGroupMember(
+      groupId,
+      targetUserId,
+    );
     if (!targetMember) {
-      throw new ForbiddenException('삭제할 멤버가 존재하지 않습니다.');
+      throw new NotFoundException('삭제할 멤버가 존재하지 않습니다.');
     }
+    await this.removeMember(targetMember.groupId, targetMember.userId);
+    return {
+      userId: targetMember.userId,
+      name: targetMember.user?.name ?? '',
+      role: targetMember.role,
+      joinedAt: targetMember.createdAt,
+    };
+  }
 
-    return targetMember;
+  async removeMember(groupId: number, userId: number): Promise<void> {
+    await this.memberRepository.deleteMember(groupId, userId);
   }
 
-  async removeMember(
-    groupId: number,
-    targetUserId: number,
-    requesterUserId: number,
-  ) {
-    const targetMember = await this.validateRemoveMember(
+  async joinGroup(
+    dto: MemberRequestDto & { userId: number },
+  ): Promise<MemberResponseDto> {
+    const { groupId, userId, role } = dto;
+    const existingMembership = await this.memberRepository.findGroupMember(
       groupId,
-      targetUserId,
-      requesterUserId,
+      userId,
     );
-
-    return this.prisma.userGroup.delete({
-      where: {
-        id: targetMember.id,
-      },
+    if (existingMembership) {
+      throw new ConflictException('이미 이 그룹에 가입되어 있습니다.');
+    }
+    const createdGroupUser = await this.memberRepository.createMember({
```

Comment:
메서드명이 joinGroup로 명명됐기때문에 joinedGroupUser가 좀 더 적절한 변수명이 아닐까 싶습니다
```suggestion
    const joinedGroupUser = await this.memberRepository.createMember({
```